File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_14_command/java/_02_after/GameStartCommand.java
package com.example._03_behavioral_patterns._14_command.java._02_after;

import com.example._03_behavioral_patterns._14_command.java._01_before.Game;

public class GameStartCommand implements Command {

    private final Game game;

    public GameStartCommand(Game game) {
        this.game = game;
    }

    @Override
    public void execute() {
        game.start();
    }

    @Override
    public void undo() {
        new GameEndCommand(this.game).execute();
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_14_command/kt/_02_after/GameStartCommand.kt
package com.example._03_behavioral_patterns._14_command.kt._02_after

import com.example._03_behavioral_patterns._14_command.kt._01_before.Game

class GameStartCommand(
    private val game: Game
) : Command {

    override fun execute() {
        game.start()
    }

    override fun undo() {
        GameEndCommand(game).execute()
    }
}
